python/cudf_polars/cudf_polars/testing/asserts.py

```diff
@@ -117,12 +121,23 @@ def assert_ir_translation_raises(q: pl.LazyFrame, *exceptions: type[Exception])
     AssertionError
        If the specified exceptions were not raised.
     """
+    translator = Translator(q._ldf.visit())
     try:
-        _ = translate_ir(q._ldf.visit())
-    except exceptions:
-        return
+        _ = translator.translate_ir()
     except Exception as e:
-        raise AssertionError(f"Translation DID NOT RAISE {exceptions}") from e
+        # The only ways an exception is raised in translation is if
+        # ...
```

Comment:
TODO: Add how exceptions are raised during translation. There are three.
1. Incompatible Polars IR version
2. Could not calculate`cudf.polars` schema. Ie. `.from_polars` raises
3. We're unable to get the current IR node or expression IR node
There are probably more, but theses are the ones I had to envelop in try-except block to get the test suite to pass.